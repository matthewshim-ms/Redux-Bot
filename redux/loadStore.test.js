const CartActions = require('./cartActions');
const DialogActions = require('./dialogActions');
const loadStore = require('./loadStore');

let session, store;

beforeEach(() => {
  session = {
    conversationData: {},
    save: jest.fn(),
    send: jest.fn()
  };

  store = loadStore(session);
});

test('"add burger" intent', () => {
  store.dispatch(DialogActions.receiveLuisIntent({
    intent: 'AddItem',
    entities: [{
      entity: 'burger',
      type: 'ItemName'
    }]
  }));

  expect(session.conversationData).toMatchSnapshot();
});
