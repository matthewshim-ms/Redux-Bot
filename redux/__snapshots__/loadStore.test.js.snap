// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"add burger" intent 1`] = `
Object {
  "cart": Object {
    "items": Array [
      "burger",
    ],
  },
}
`;
